spring-kafka/src/test/java/org/springframework/kafka/listener/ConcurrentMessageListenerContainerTests.java

```diff
@@ -87,17 +90,18 @@
 
 	@Test
 	public void testAutoCommit() throws Exception {
-		logger.info("Start auto");
+		this.logger.info("Start auto");
 		Map<String, Object> props = KafkaTestUtils.consumerProps("test1", "true", embeddedKafka);
 		DefaultKafkaConsumerFactory<Integer, String> cf = new DefaultKafkaConsumerFactory<Integer, String>(props);
 		ConcurrentMessageListenerContainer<Integer, String> container =
 				new ConcurrentMessageListenerContainer<>(cf, topic1);
 		final CountDownLatch latch = new CountDownLatch(4);
+		final ConcurrentSkipListSet<String> listenerThreadNames = new ConcurrentSkipListSet<>();
```

Comment:
Well, according to your assertion that can be just simple (unsychronized) `ArrayList`.
OK. That's just nit-pick and fully isn't important in favor of all that hard work you did here.
Never mind